github/git_refs_test.go

```diff
@@ -56,6 +56,154 @@ func TestGitService_GetRef(t *testing.T) {
+		  {
+		    "ref": "refs/heads/b",
+		    "url": "https://api.github.com/repos/o/r/git/refs/heads/b",
+		    "object": {
+		      "type": "commit",
+		      "sha": "aa218f56b14c9653891f9e74264a383fa43fefbd",
+		      "url": "https://api.github.com/repos/o/r/git/commits/aa218f56b14c9653891f9e74264a383fa43fefbd"
+		    }
+		  }`)
+	})
+
+	refs, _, err := client.Git.GetRefs(context.Background(), "o", "r", "refs/heads/b")
+	if err != nil {
+		t.Fatalf("Git.GetRefs returned error: %v", err)
+	}
+
+	ref := refs[0]
+	want := &Reference{
+		Ref: String("refs/heads/b"),
+		URL: String("https://api.github.com/repos/o/r/git/refs/heads/b"),
+		Object: &GitObject{
+			Type: String("commit"),
+			SHA:  String("aa218f56b14c9653891f9e74264a383fa43fefbd"),
+			URL:  String("https://api.github.com/repos/o/r/git/commits/aa218f56b14c9653891f9e74264a383fa43fefbd"),
+		},
+	}
+	if !reflect.DeepEqual(ref, want) {
+		t.Errorf("Git.GetRefs returned %+v, want %+v", ref, want)
+	}
+
+	// without 'refs/' prefix
+	if _, _, err := client.Git.GetRefs(context.Background(), "o", "r", "heads/b"); err != nil {
+		t.Errorf("Git.GetRefs returned error: %v", err)
+	}
+}
+
+func TestGitService_GetRefs_multipleRefs(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/git/refs/heads/b", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		fmt.Fprint(w, `
+		  [
+		    {
+			    "ref": "refs/heads/booger",
+			    "url": "https://api.github.com/repos/o/r/git/refs/heads/booger",
+			    "object": {
+			      "type": "commit",
+			      "sha": "aa218f56b14c9653891f9e74264a383fa43fefbd",
+			      "url": "https://api.github.com/repos/o/r/git/commits/aa218f56b14c9653891f9e74264a383fa43fefbd"
+			    }
+		  	},
+		    {
+		      "ref": "refs/heads/bandsaw",
+		      "url": "https://api.github.com/repos/o/r/git/refs/heads/bandsaw",
+		      "object": {
+		        "type": "commit",
+		        "sha": "612077ae6dffb4d2fbd8ce0cccaa58893b07b5ac",
+		        "url": "https://api.github.com/repos/o/r/git/commits/612077ae6dffb4d2fbd8ce0cccaa58893b07b5ac"
+		      }
+		    }
+		  ]
+		`)
+	})
+
+	refs, _, err := client.Git.GetRefs(context.Background(), "o", "r", "refs/heads/b")
+	if err != nil {
+		t.Errorf("Git.GetRefs returned error: %v", err)
+	}
+
+	want := &Reference{
+		Ref: String("refs/heads/booger"),
+		URL: String("https://api.github.com/repos/o/r/git/refs/heads/booger"),
+		Object: &GitObject{
+			Type: String("commit"),
+			SHA:  String("aa218f56b14c9653891f9e74264a383fa43fefbd"),
+			URL:  String("https://api.github.com/repos/o/r/git/commits/aa218f56b14c9653891f9e74264a383fa43fefbd"),
+		},
+	}
+	if !reflect.DeepEqual(refs[0], want) {
+		t.Errorf("Git.GetRefs returned %+v, want %+v", refs[0], want)
+	}
+}
+
+func TestGitService_GetRefs_noRefs(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/git/refs/heads/b", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		fmt.Fprint(w, "[]")
```

Comment:
Sounds good. My only request is that you document that this test is for an _unexpected_ response from GitHub. It should never actually happen.

My concern is that if we don't document that, it will be confusing later.

You could also check that the error string is an exact match with "unexpected response from GitHub API: an array of refs with length 0". That way it ties the test closer with the exact thing it's testing.